src/bepasty/apis/rest.py

```diff
@@ -171,77 +172,52 @@ def get(self):
         if not may(LIST):
             return 'Missing Permissions', 403
         ret = {}
-        for name in current_app.storage:
-            item = current_app.storage.open(name)
+        for meta in file_infos():
+            name = meta.pop(ID)
             ret[name] = {'uri': url_for('bepasty_apis.items_detail', name=name),
-                         'file-meta': dict(item.meta)}
+                         'file-meta': meta}
         return jsonify(ret)
 
 
-class ItemDetailView(RestBase):
-    @rest_errorhandler
-    def get(self, name):
-        if not may(READ):
-            return 'Missing Permissions', 403
-
-        with current_app.storage.open(name) as item:
-            return jsonify({'uri': url_for('bepasty_apis.items_detail', name=name),
-                            'file-meta': dict(item.meta)})
+class ItemDetailView(DownloadView, RestBase):
+    def err_incomplete(self, item, error):
+        raise Conflict(description=error)
 
-
-class ItemDownloadView(RestBase):
-    content_disposition = 'attachment'
+    def response(self, item, name):
+        return jsonify({'uri': url_for('bepasty_apis.items_detail', name=name),
+                        'file-meta': dict(item.meta)})
 
     @rest_errorhandler
     def get(self, name):
-        if not may(READ):
-            return 'Missing Permissions', 403
+        return super(ItemDetailView, self).get(name)
 
-        try:
-            item = current_app.storage.open(name)
-        except (OSError, IOError) as e:
-            if e.errno == errno.ENOENT:
-                return 'File not found', 404
-            raise
-
-        if item.meta.get(LOCKED):
-            error = 'File Locked.'
-        elif not item.meta.get(COMPLETE):
-            error = 'Upload incomplete. Try again later.'
-        else:
-            error = None
-        if error:
-            item.close()
-            return error, 403
 
+class ItemDownloadView(ItemDetailView):
+    def response(self, item, name):
         request_range = DownloadRange.from_request()
         if not request_range:
-            range_end = item.data.size - 1
-            range_begin = 0
+            size = item.data.size
+            start = 0
         else:
             if request_range.end == -1:
-                range_end = item.data.size - 1
+                size = item.data.size
```

Comment:
guess this needs to consider request_range.begin

Reply:
in general, if the old way of dealing with begin/end was **not** broken, i'ld prefer you to not change it, but just keep it as is.

Reply:
yes, usually I dont change. but this time self.stream() was consolidated and was changed to handle web site.  so -1 and +1 was repeated, and unreadable for me.

anyway, reverted unnecessary changes.